pkg/certificates/config.go

```diff
@@ -129,7 +130,95 @@ func ValidateComponentWithConfig(component string, config *RenewalConfig) error
 	return nil
 }
 
-// shouldProcessComponent checks if the specified component should be processed.
-func shouldProcessComponent(requestedComponent, targetComponent string) bool {
-	return requestedComponent == "" || requestedComponent == targetComponent
+func PopulateConfig(ctx context.Context, cfg *RenewalConfig, kubeClient kubernetes.Client, cluster *types.Cluster) error {
+
+	if len(cfg.ControlPlane.Nodes) > 0 {
+		return nil
+	}
+
+	controlPlaneIPs, err := getControlPlaneIPs(ctx, kubeClient, cluster)
+	if err != nil {
+		return fmt.Errorf("cluster is not reachable please provide control plane and/or external etcd IP addresses: %w", err)
+	}
+
+	cfg.ControlPlane.Nodes = controlPlaneIPs
+
+	etcdIPs, err := getEtcdIPs(ctx, kubeClient, cluster)
+	if err != nil {
+		fmt.Printf("Warning: Failed to get etcd IPs: %v\n", err)
+	} else {
+		cfg.Etcd.Nodes = etcdIPs
+	}
+
+	return nil
+}
+
+func getControlPlaneIPs(ctx context.Context, kubeClient kubernetes.Client, cluster *types.Cluster) ([]string, error) {
+	fmt.Printf("getControlPlaneIPs \n")
+	var controlPlaneIPs []string
+
+	machineList := &clusterv1.MachineList{}
+
+	namespaceOpt := kubernetes.ListOptions{
+		Namespace: constants.EksaSystemNamespace,
+	}
+
+	if err := kubeClient.List(ctx, machineList, namespaceOpt); err != nil {
+		return nil, fmt.Errorf("listing machines: %w", err)
+	}
+
+	for _, machine := range machineList.Items {
+		if machine.Labels["cluster.x-k8s.io/cluster-name"] == cluster.Name {
```

Comment:
Can we use clusterNameLabel and controlPlaneLabel variable for these labels?
Also, Can we pass these labels as a selector when we make kubeClient.List? Same for external etcd.